### DISCUSSION TITLE
LIS Testcases DA vague error

### DISCUSSION THREAD

**Discussion by dechowjack** [NASA-LIS/LISF#896] [Q&A] [answered]
Hello, our lab has been working to incorporate LIS into our modeling. We have successfully run standard LIS at a variety of resolutions. I have been attempting to work through the data assimilation test cases in order to understand the process better so that we can incorporate it into our work. Okay, now for the error.

Running on NASA Discover, I have successfully worked through test cases 1 through 5. While working through testcase 6, I encounter a vague error. Consulting the log files, it appears that every piece of input gets successfully read by LIS. The first error is a warning that states 

[WARN] DA obs domain details
[WARN] Error in nf90_get_att: GRIDTYPE

The second error is denoted with [ERR]

[ERR] SMAP(NASA) soil moisture Composite Release ID: is missing Stopping.
[ERR] endrun is being called

Has anyone experienced these errors before? If so, has anyone found a solution to these errors?

Best

Jack

**Comment by dmocko** [comment]
Hello Jack,

The first warning should actually appear in your log files as:

 [INFO] DA obs domain details:
 [WARN] ****************WARNING*********************
 [WARN] Error in nf90_get_att: GRIDTYPE
 [WARN] ****************WARNING*********************

If you are not seeing exactly what I have above, please let us know.

This warning is not a concern - it's simply letting the user know that the DA obs domain netCDF file does not have the GRIDTYPE attribute.  This warning will not affect the run (although the LIS team should fix this to ensure that the DA obs domain files are written with this attribute).

The second error is stopping the run.  The testcase suite is designed to work with the specific GitHub commit ID on the bottom of this page:
https://lis.gsfc.nasa.gov/lis-testcases

You may be using a more recent or current version of the software, which is looking for this missing config entry in the lis.config file.  New code added since that commit ID is looking for this config entry.

If you look at the LIS Users' Guide (best viewed in the browser as detail on this page): https://github.com/NASA-LIS/LISF/tree/master/docs

You can see that there is an explanation of this config entry.  Please add it to your lis.config and try again.  You may be using R17 instead of R16 - please check the filenames of the data you are using on Discover.

SMAP(NASA) soil moisture Composite Release ID: specifies the SMAP NASA composite release ID included within the filename (e.g., R16).

David

**Reply by bmcandr** [reply]
I am able to run step 6 of the public testcase with LIS compiled from the current master branch. We updated the testcase files last summer which involved the addition of the `SMAP(NASA) soil moisture Composite Release ID:` entry to the LDT and LIS config files. This leads me to suspect that the testcase files being used are out of date...

@dechowjack, can you check that you are using the latest version of the [testcase files](https://lis.gsfc.nasa.gov/lis-testcases)? The latest version of the tar files end in `_2020.tar.gz`.

As @dmocko mentioned, you'll have the best luck running the public testcase with LDT/LIS/LVT compiled from the commit indicated on the testcase website. See the "Setup" section in this testcase walkthrough doc I created for instructions: https://github.com/bmcandr/lis-public-tc-walkthrough

**Reply by dechowjack** [reply]
@dmocko Hi David, the first warning does indeed appear that way in my log files, I just left out the warnings when I typed it up. Second, I went into the config file and the composite ID is specified as R16. When I look at the SMAP files, they have the name format of 

SMAP_L3_SM_P_20170301_R16010_001.h5

Does this mean that the ID is actually 16.xxx? Or should the R16 ID be enough? 

@bmcandr Hi Brandon, I am indeed working from the 2020 gzip files.

**Reply by dmocko** [reply]
@dechowjack You should just use "R16" (without the quotes) for this config entry.

You may have a typo, or extra space, or change in capitalization in your lis.config file.  Can you please double check?

LIS is unable to find this exact phrase in the lis.config file:
----
SMAP(NASA) soil moisture Composite Release ID:  R16
----

David

**Reply by dmocko** [reply]
Sorry for the weird random bolding.  I hate copy and paste most of the time....

**Reply by dechowjack** [reply]
I just checked, it is identical to the line you posted.

**Comment by LIS-navari** [comment]
Hi,

If you are using the new SMAP data you have to use R17.
Please look at the name of the product to figure out the correct Composite Release ID.
e.g.,
SMAP_L3_SM_P_20210909_R17030_001.h5

Thanks
Mahdi
From: David Mocko ***@***.***>
Reply-To: NASA-LIS/LISF ***@***.***>
Date: Wednesday, September 15, 2021 at 12:01 PM
To: NASA-LIS/LISF ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [EXTERNAL] Re: [NASA-LIS/LISF] LIS Testcases DA vague error (Discussion #896)


@dechowjack<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fdechowjack&data=04%7C01%7Cmahdi.navari%40nasa.gov%7C74106b583e0d46bc006808d978620e19%7C7005d45845be48ae8140d43da96dd17b%7C0%7C0%7C637673184815643870%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=B3KNRRWzxLWCj3FjiGat27Vh6jCOOcurZ%2BrRzRldV%2FE%3D&reserved=0> You should just use "R16" (without the quotes) for this config entry.

You may have a typo, or extra space, or change in capitalization in your lis.config file. Can you please double check?

LIS is unable to find this exact phrase in the lis.config file:
SMAP(NASA) soil moisture Composite Release ID: R16

David

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FNASA-LIS%2FLISF%2Fdiscussions%2F896%23discussioncomment-1334275&data=04%7C01%7Cmahdi.navari%40nasa.gov%7C74106b583e0d46bc006808d978620e19%7C7005d45845be48ae8140d43da96dd17b%7C0%7C0%7C637673184815653823%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=fUPXJI%2Bi%2BPXlYPsjGCNrmGVGhUf%2BYksHlAekU1v5V%2BM%3D&reserved=0>, or unsubscribe<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAKNKXD4RRHMV4EPB6ONNYNTUCC7MXANCNFSM5EAYQ5JA&data=04%7C01%7Cmahdi.navari%40nasa.gov%7C74106b583e0d46bc006808d978620e19%7C7005d45845be48ae8140d43da96dd17b%7C0%7C0%7C637673184815653823%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=FABZXcnSC7%2FyDW%2FKLCsdqGZvANRt3LDiCtJ2F8OHkBk%3D&reserved=0>.
Triage notifications on the go with GitHub Mobile for iOS<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapps.apple.com%2Fapp%2Fapple-store%2Fid1477376905%3Fct%3Dnotification-email%26mt%3D8%26pt%3D524675&data=04%7C01%7Cmahdi.navari%40nasa.gov%7C74106b583e0d46bc006808d978620e19%7C7005d45845be48ae8140d43da96dd17b%7C0%7C0%7C637673184815663770%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=z%2Bm4uRtdctvPGWFvbqfOpw8Vfj6XEsI6Vzf5IJ43GmE%3D&reserved=0> or Android<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Dcom.github.android%26referrer%3Dutm_campaign%253Dnotification-email%2526utm_medium%253Demail%2526utm_source%253Dgithub&data=04%7C01%7Cmahdi.navari%40nasa.gov%7C74106b583e0d46bc006808d978620e19%7C7005d45845be48ae8140d43da96dd17b%7C0%7C0%7C637673184815663770%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=J05W5gPXymIqEPnE3YBzE28C6oIji9iOjz8XyPI4qgo%3D&reserved=0>.

**Comment by dechowjack** [comment]
@dmocko I've been working on installing/compiling the LIS framework with the same commit ID as the public testcases so I can rerun the test cases. While doing so, I ran into an error installing one of the dependencies (OpenJPG) requiring sudo access to finish the install. I have tried to install in both home and $nobackup directory. Do you have any suggestions on what to do?

**Reply by bmcandr** [reply]
@dechowjack, you mentioned that you're working on Discover. We have all of the necessary libraries already installed and you should be able to use the module files under `LISF/env/discover` to setup your environment to use those.

I have some instructions here for setting up your environment on Discover: https://github.com/bmcandr/lis-bootcamp/blob/master/docs/LIS-on-NCCS-discover.md#setting-up-your-environment-on-discover

**Comment by dmocko** [comment]
Thanks, @bmcandr - I was just going to suggest the same.

@dechowjack - I e-mailed you yesterday at your OSU address about you sharing your Discover directory, so I can take a look to see why you are getting that error.  Did you get that e-mail?

David

**Comment by dmocko** [comment]
@dechowjack - Thanks for sharing your directory to me via e-mail.  I can't see anything wrong with it, so it may be that your libraries are causing a conflict.  Can you please try the modulefile setup method, and use only those libraries to compile?  Be sure you don't have other modules loaded when you compile/run, as this can cause a conflict.

**Comment by dechowjack** [comment, marked as answer]
@dmocko @bmcandr @LIS-navari 

Hi All,

After recompiling the most recent commit of LIS, I was able to successfully run the data assimilation test case with all of my previous inputs. The only thing that I changed was the compiled LIS executable in the folder. Thanks for all the help.

Best

Jack
